packages/contracts/deploy/new/40_finalize-management-dao/20_register-management-dao-on-dao-registry.ts

```diff
@@ -73,10 +86,21 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     network.name
   );
 
-  const setMetadataTX = await managementDaoContract.setMetadata(
-    ethers.utils.hexlify(ethers.utils.toUtf8Bytes(`ipfs://${metadataCIDPath}`))
+  const hasMetadataPermission = await managementDaoContract.hasPermission(
+    managementDaoContract.address,
+    deployer.address,
+    ethers.utils.id('SET_METADATA_PERMISSION'),
+    '0x'
   );
-  await setMetadataTX.wait();
+
+  if (hasMetadataPermission) {
```

Comment:
What makes this `if` necessary? I think we know already that the `deployer` has it from `packages/contracts/deploy/new/40_finalize-management-dao/00_grant-permissions.ts` or am I missing something?